package/kedro_viz/data_access/managers.py

```diff
@@ -316,7 +318,17 @@ def add_dataset(
         Returns:
             The GraphNode instance representing the dataset that was added to the NodesRepository.
         """
-        obj = self.catalog.get_dataset(dataset_name)
+        try:
+            obj = self.catalog.get_dataset(dataset_name)
+        except DatasetError:
+            # This is to handle dataset factory patterns when running
+            # Kedro Viz in lite mode. The `get_dataset` function
+            # of DataCatalog calls AbstractDataset.from_config
+            # which tries to create a Dataset instance from the pattern
+
+            # pylint: disable=abstract-class-instantiated
+            obj = MemoryDataset()  # type: ignore[abstract]
```

Comment:
For now, this approach works, but in the future, we may need to evaluate whether to represent this as a `MemoryDataset` or as a new term like  `UnavailableDataset` (or similar). The latter would specifically denote a dataset that cannot be loaded due to missing dependencies

Reply:
Do we need to do this here, as `get_dataset` handles `DatasetNotFoundError` with `MemoryDataset` anyway?

Reply:
also would it make sense to define catalog as a Union[DatasetCatalog, DatasetCatalogLite] -- i am not completely sure how much it would make the code better .

Reply:
> we may need to evaluate whether to represent this as a MemoryDataset or as a new term like UnavailableDataset (or similar). This would specifically denote a dataset that cannot be loaded due to missing dependencies

Yes I wanted to have a discussion and we can handle this here - https://github.com/kedro-org/kedro-viz/issues/2058

> Do we need to do this here, as get_dataset handles DatasetNotFoundError with MemoryDataset anyway?
Also do we need to add the MemoryDataset logic to _get_dataset in DataCatalogLite?

So the catalog here is DataCatalog instance and not DataCatalogLite. We can either convert the CatalogRepository to hold DataCatalogLite instance or handle it here. I felt since we have 2 paths - normal and lite, for now we can handle this here which is much cleaner way. In future if we decide to go with DataCatalogLite for Kedro-Viz as a normal path, we can make the change. Wdyt ?